taichi/program/compile_config.h

```diff
@@ -42,6 +42,7 @@ struct CompileConfig {
   bool make_block_local;
   bool detect_read_only;
   bool ndarray_use_torch;
+  bool ndarray_use_cached_allocator;
```

Comment:
Do we need this additional flag?

Reply:
Just to have an option to turn the caching behavior off. It might be useful later for profiling.

Reply:
IMHO caching is strictly better than the one without, so we probably can just show some perf comparison in this PR and then save us from adding a new compile config (and the branches). But I'm pretty open on this. :D